src/main/webapp/dev/DatePickerJsTest.js

```diff
@@ -0,0 +1,26 @@
+QUnit.module('datepicker.js');
+
+QUnit.test('getMinDateForEndDate(startDate)', function(assert) {
+    assert.deepEqual(getMinDateForEndDate(new Date('02/25/2014')), new Date('02/25/2014'), 'Return startDate');
+});
+
+QUnit.test('getMaxDateForStartDate(endDate)', function(assert) {
+    assert.deepEqual(getMaxDateForStartDate(new Date('02/04/2014')), new Date('02/04/2014'), 'Return endDate');
+});
+
+QUnit.test('getMinDateForPublishDate(visibleDate)', function(assert) {
+    assert.deepEqual(getMinDateForPublishDate(new Date('10/13/2014')), new Date('10/13/2014'), 'Return visibleDate');
+});
+
+QUnit.test('getMaxDateForVisibleDate(startDate,publishDate)', function(assert) {
+    assert.deepEqual(getMaxDateForVisibleDate(new Date('08/25/2014'), null), new Date('08/25/2014'),
```

Comment:
@Mynk96 Sorry I may not make this clear.
The thing is that when we write test cases, we want write effective test cases (obviously, we cannot write full test cases for the function as the `date` is kind of infinity). `null` value is considered as a special input so that we want to test it more. You have `null` value for `publishDate`, so we also need `null` value for `startDate` or both.